chain/chain/src/flat_storage_resharder.rs

```diff
@@ -1052,19 +1095,11 @@ pub enum TaskExecutionStatus {
     NotStarted,
 }
 
-/// Result of a simple flat storage resharding task.
+/// Result of a schedulable flat storage resharding task.
 #[derive(Clone, Debug, Copy, Eq, PartialEq)]
 pub enum FlatStorageReshardingTaskResult {
```

Comment:
I think there might still be value in having different task results for split shard and catchup, because catchup can't be cancelled AFAIK

Something like:
split -> FlatStorageReshardingCancellableTaskResult
catchup -> FlatStorageReshardingTaskResult

Reply:
So looks like the cancellation comes just from [receiving SIGINT](https://github.com/near/nearcore/blob/beb00f4aa1b344a196012b9357b02e21cf167aab/neard/src/cli.rs#L588)? In which case it should be possible to get a cancellation during catchup too, it's just that it was never checked before. Added a check at the top of `shard_catchup_task()` returning `FlatStorageReshardingTaskResult::Cancelled` if `stop()` has been called. Let me know what you think though

Reply:
Yeah that's actually quite good, it is better to be able to interrupt the catchup task as well. On this point, you could add the check also inside the loop in `shard_catchup_apply_deltas` to interrupt there and exit the loop early.

Comment on line 64 will need to be adjusted

```
///     - Children shard catchup is a consequence of splitting a shard, not a resharding event on
///       its own. As such, it can't be manually cancelled.
```

to something like


```
- Children shard catchup can be cancelled and will resume from the point where it left.
```

Reply:
whoops forgot about this one, updated comment. For now just left a TODO to check it in the loop, and can revisit it in another PR, since it needs a bit of refactoring of the return value of that function, and this PR is already a little too big and kinda changing too much.